### DISCUSSION TITLE
Fatal error on /var/lib/rpm

### DISCUSSION THREAD

**Discussion by Viengogo** [aquasecurity/trivy#4892] [Q&A]
### Question

We using Trivy to scan fs, sometimes on some scans we have fatal errors on the /var/lib/rpm directory due to corrupt databases. 

We could skip this directory /var/lib/rpm but doesn't this exclusion restrict trivy for CVE detection on OS packages?

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

**Comment by knqyf263** [comment]
Trivy depends on `/var/lib/rpm` to obtain the installed packages. if you skip it, Trivy will not find OS packages. I think you have to fix the database.
https://access.redhat.com/solutions/6903

**Reply by fredericg78** [reply]
Thanks for this answer. We cannot fix the rpm databases as they don't belong to us, only the VM owners can do it (we can just signal owners they have to do it). The main issue we have about trivy is that it stops itself when confronted with a corrupted rpm db; instead of that, we'd like to see trivy to continue the filesystem scan and to just ouput a warning about the rpm issue, so we could have at the end a partial report instead of nothing with a FATAL error.

**Reply by knqyf263** [reply]
You can skip the database with `--skip-files`.

**Reply by fredericg78** [reply]
Or the rpm directory with --skip-dirs. Of course, but we can't ! We launch trivy in client/server mode over thousands of VM (we scan an entire cloud), we don't know by advance if a VM will have a rpm corrupted db. Thats's why we want trivy to scan the entire VM filesystem and to signal in its report if it's incomplete on some directories, just by warnings.

**Reply by knqyf263** [reply]
BTW, I think it's already skipped. Does panic occur? Could you share the error message and how to reproduce it?

**Reply by Viengogo** [reply]
We having this error when database are corrupt :
error: db5 error(-30973) from dbenv->failchk: BDB0087 DB_RUNRECOVERY: Fatal error, run database recovery
scheduler-worker-scan
error: cannot open Packages database in /var/lib/rpm

When it occurs, trivy stopped to scan

**Reply by knqyf263** [reply]
Would you mind creating such a container image and sharing it with us?
